### PR TITLE
[Keyboard] NCR-80

### PR DESCRIPTION
**Pull request by peepeetee**
## Description

What I did:
Changed around the product ID to match the vendor's files
Added status LED defines
Added some keymaps for the different configs this board could have
Made the appropriate layout macros for LAYOUT_tkl_ansi and LAYOUT_tkl_iso

TODO:
- [x] Make the layout macros for LAYOUT_tkl_ansi_wkl and LAYOUT_tkl_iso_wkl
- [x] Make it so that keymaps ~~have~~ does not have RGB control
- [x] Add the hotswap version
- [ ] ?????????
- [ ] Profit

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mt/ncr80/soldered/rules.mk [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# Build Options
+BOOTMAGIC_ENABLE = yes     # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = yes      # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+NKRO_ENABLE = yes          # Enable N-Key Rollover
+BACKLIGHT_ENABLE = yes     # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no      # Enable Bluetooth
```
Comment (drashna) [inline keyboards/mt/ncr80/soldered/rules.mk:16]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
These need to be addressed: 
☒ mt/ncr80/soldered: Claims to support a community layout that does not exist: tkl_ansi_wkl
☒ mt/ncr80/soldered: Claims to support community layout tkl_iso_wkl but no LAYOUT_tkl_iso_wkl() macro found

File: keyboards/mt/ncr80/soldered/rules.mk [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = yes      # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+NKRO_ENABLE = yes          # Enable N-Key Rollover
+BACKLIGHT_ENABLE = yes     # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no       # Enable keyboard RGB underglow
+AUDIO_ENABLE = no          # Audio output
+
+LAYOUTS = tkl_ansi tkl_iso tkl_ansi_wkl tkl_iso_wkl
```
Comment (drashna) [inline keyboards/mt/ncr80/soldered/rules.mk:18]: `tkl_ansi_wkl` `tkl_iso_wkl` are not supported community layouts, currently.

```suggestion
LAYOUTS = tkl_ansi tkl_iso
```

File: keyboards/mt/ncr80/soldered/keymaps/default_tkl_iso/keymap.c [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+};
+
+
+    //   _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,             _______,  _______,  _______, \
+    //   _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,   _______,  _______,  _______, \
+    //   _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,             _______,  _______,  _______, \
+    //   _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______, \
+    //   _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,                       _______, \
+    //   _______,  _______,  _______,                _______,                                 _______,_______, _______,  _______,                      _______, _______, _______  )
```
Comment (fauxpark) [inline keyboards/mt/ncr80/soldered/keymaps/default_tkl_iso/keymap.c:36]:
```suggestion
  [0] = LAYOUT_tkl_iso(
      KC_ESC, KC_F1,  KC_F2,  KC_F3,  KC_F4,  KC_F5,  KC_F6,  KC_F7,  KC_F8,  KC_F9,  KC_F10, KC_F11,  KC_F12,             KC_PSCR, KC_SLCK, KC_PAUS,
      KC_GRV, KC_1,   KC_2,   KC_3,   KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,   KC_0,   KC_MINS, KC_EQL,  KC_BSPC,   KC_INS,  KC_HOME, KC_PGUP,
      KC_TAB, KC_Q,   KC_W,   KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,   KC_P,   KC_LBRC, KC_RBRC,            KC_DEL,  KC_END,  KC_PGDN,
      KC_CAPS,KC_A,   KC_S,   KC_D,   KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,   KC_SCLN,KC_QUOT, KC_NUHS, KC_ENT,
      KC_LSFT,KC_NUBS,KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM,KC_DOT, KC_SLSH,          KC_RSFT,            KC_UP,
      KC_LCTL,KC_LGUI,KC_LALT,                KC_SPC,                                 KC_RALT,KC_RGUI, KC_APP,  KC_RCTL,   KC_LEFT, KC_DOWN, KC_RGHT  ),

};
```

File: keyboards/mt/ncr80/readme.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# NCR-80
```
Comment (fauxpark) [inline keyboards/mt/ncr80/readme.md:1]: Needs to follow the readme template.

File: keyboards/mt/ncr80/hotswap/rules.mk [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
```
Comment (fauxpark) [inline keyboards/mt/ncr80/hotswap/rules.mk:7]:
```suggestion
# Build Options
#   change yes to no to disable
#
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mt/ncr80/readme.md (hunk 4 above)
Comment (peepeetee) [reply keyboards/mt/ncr80/readme.md:1]: This is the top level readme, both readmes under soldered and hotswap follow the template, and my thought process is that, well, as written, 

>For more information, read the readme.md in soldered/ or hotswap/, depending on the version you are interested in.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/mt/ncr80/hotswap/hotswap.h [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "quantum.h"
+
+#define XXX KC_NO
+
+
+#define LAYOUT( \
```
Comment (fauxpark) [inline keyboards/mt/ncr80/hotswap/hotswap.h:23]:
```suggestion
#define LAYOUT_tkl_ansi_wkl( \
```

File: keyboards/mt/ncr80/hotswap/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT(
```
Comment (fauxpark) [inline keyboards/mt/ncr80/hotswap/keymaps/default/keymap.c:20]:
```suggestion
  [0] = LAYOUT_tkl_ansi_wkl(
```

File: keyboards/mt/ncr80/hotswap/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT(
```
Comment (fauxpark) [inline keyboards/mt/ncr80/hotswap/keymaps/via/keymap.c:20]:
```suggestion
  [0] = LAYOUT_tkl_ansi_wkl(
```

File: keyboards/mt/ncr80/hotswap/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,    KC_F10,     KC_F11,     KC_F12,
+    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,     KC_MINS,    KC_EQL,     KC_BSPC,                 KC_INS,  KC_HOME, KC_PGUP,
+    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,     KC_LBRC,    KC_RBRC,    KC_BSLS,                 KC_DEL,  KC_END,  KC_PGDN,
+    KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN,  KC_QUOT,                KC_ENT,
+    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,  KC_RSFT,                                                  KC_UP,
+    KC_LCTL, KC_LALT,                   KC_SPC,                                      KC_RALT,           KC_RCTL,                                         KC_LEFT, KC_DOWN, KC_RGHT),
+  [1] = LAYOUT(
```
Comment (fauxpark) [inline keyboards/mt/ncr80/hotswap/keymaps/via/keymap.c:27]:
```suggestion
  [1] = LAYOUT_tkl_ansi_wkl(
```

File: keyboards/mt/ncr80/hotswap/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    _______,  _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,    _______,     _______,     _______,
+    _______,  _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,     _______,    _______,     _______,                 _______,  _______, _______,
+    _______,  _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,     _______,    _______,    _______,                 _______,  _______,  _______,
+    _______, _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,  _______,                _______,
+    _______, _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______, _______,  _______,  _______,                                                  _______,
+    _______, _______,                   _______,                                      _______,           _______,                                         _______, _______, _______),
+  [2] = LAYOUT(
```
Comment (fauxpark) [inline keyboards/mt/ncr80/hotswap/keymaps/via/keymap.c:34]:
```suggestion
  [2] = LAYOUT_tkl_ansi_wkl(
```

File: keyboards/mt/ncr80/hotswap/keymaps/via/keymap.c [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+    _______,  _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,    _______,     _______,     _______,
+    _______,  _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,     _______,    _______,     _______,                 _______,  _______, _______,
+    _______,  _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,     _______,    _______,    _______,                 _______,  _______,  _______,
+    _______, _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,  _______,                _______,
+    _______, _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______, _______,  _______,  _______,                                                  _______,
+    _______, _______,                   _______,                                      _______,           _______,                                         _______, _______, _______),
+  [3] = LAYOUT(
```
Comment (fauxpark) [inline keyboards/mt/ncr80/hotswap/keymaps/via/keymap.c:41]:
```suggestion
  [3] = LAYOUT_tkl_ansi_wkl(
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mt/ncr80/hotswap/hotswap.h (hunk 6 above)
Comment (peepeetee) [reply keyboards/mt/ncr80/hotswap/hotswap.h:23]: I don't think this applies, as um the keyboard is missing prtsc, scrlk, and pause

Comment (fauxpark) [reply keyboards/mt/ncr80/hotswap/hotswap.h:23]: Then the info.json is wrong.

Comment (peepeetee) [reply keyboards/mt/ncr80/hotswap/hotswap.h:23]: Noted and corrected

**Comment by Xemuel** [comment]
Any update on this?

**Comment by peepeetee** [comment]
> Any update on this?

I think it's just waiting for reviews

File: keyboards/mt/ncr80/hotswap/hotswap.h (hunk 6 above)
Comment (fauxpark) [reply keyboards/mt/ncr80/hotswap/hotswap.h:23]: These are still outstanding.

Comment (peepeetee) [reply keyboards/mt/ncr80/hotswap/hotswap.h:23]: Could I get some pointers on what I should be doing to the PR at this point? I'm a bit out of the loop and needs some reorientation

Comment (fauxpark) [reply keyboards/mt/ncr80/hotswap/hotswap.h:23]: Look at the reviews.

Comment (peepeetee) [reply keyboards/mt/ncr80/hotswap/hotswap.h]: aight done thanks

**Comment by drashna** [comment]
> ☒ mt/ncr80/hotswap: Number of keys for LAYOUT_tkl_ansi_wkl does not match! info.json specifies 84 keys, C macro specifies 81

File: keyboards/mt/ncr80/hotswap/readme.md [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+
+![NCR-80](https://i.imgur.com/kAjbAPLl.jpg)
+
+> Re-imagination of the classic rebranded Cherry G81-3000 by NCR
+
+NCR-80 is an affordable keyboard kit inspired by iconic vintage keyboards.
+
+* Keyboard Maintainer: QMK Community
+* Hardware Supported: ATmega32U4
+* Hardware Availability: [AliExpress](https://www.aliexpress.com/item/1005003345941543.html) [淘宝](https://item.taobao.com/item.htm?id=657482028672)
```
Comment (noroadsleft) [inline keyboards/mt/ncr80/hotswap/readme.md:11]:
```suggestion
* Hardware Availability: [AliExpress](https://www.aliexpress.com/item/1005003345941543.html), [Taobao 淘宝](https://item.taobao.com/item.htm?id=657482028672)
```

File: keyboards/mt/ncr80/hotswap/readme.md [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader
+
+Enter the bootloader in 3 ways:
+
+* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
+* **Physical reset button**: Briefly press the button on the back of the PCB
+* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```
Comment (noroadsleft) [inline keyboards/mt/ncr80/hotswap/readme.md:29]:
```suggestion
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB
* **Keycode in layout**: Press the key mapped to `RESET` if it is available

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/mt/ncr80/readme.md [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+NCR-80 is an affordable keyboard kit inspired by iconic vintage keyboards.
+
+For more information, read the readme.md in soldered/ or hotswap/, depending on the version you are interested in.
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ncr80/soldered:default
+    make ncr80/hotswap:default
+
+
```
Comment (noroadsleft) [inline keyboards/mt/ncr80/readme.md:16]:
```suggestion
For more information, read the readme.md in [soldered/](soldered/) or [hotswap/](hotswap/), depending on the version you are interested in.
```

File: keyboards/mt/ncr80/hotswap/info.json [hunk 15]
```diff
@@ -0,0 +1,95 @@
+{
+    "keyboard_name": "NCR-80",
```
Comment (noroadsleft) [inline keyboards/mt/ncr80/hotswap/info.json:2]:
```suggestion
    "keyboard_name": "NCR-80 Hotswap",
```

File: keyboards/mt/ncr80/soldered/info.json [hunk 16]
```diff
@@ -0,0 +1,284 @@
+{
+    "keyboard_name": "NCR-80",
+    "url": "https://www.aliexpress.com/item/1005003345941543.html",
```
Comment (noroadsleft) [inline keyboards/mt/ncr80/soldered/info.json:3]:
```suggestion
    "keyboard_name": "NCR-80 Soldered",
    "url": "https://www.aliexpress.com/item/1005003364462523.html",
```

File: keyboards/mt/ncr80/soldered/readme.md [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader
+
+Enter the bootloader in 3 ways:
+
+* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
+* **Physical reset button**: Briefly press the button on the back of the PCB
+* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```
Comment (noroadsleft) [inline keyboards/mt/ncr80/soldered/readme.md:29]:
```suggestion
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB
* **Keycode in layout**: Press the key mapped to `RESET` if it is available

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Pull request closed without merging**